panic on error

A user reported that when restic 0.3.1 encounters an error during backup, it panics:
```
scanned 34315 directories, 130418 files in 0:15
panic: runtime error: invalid memory address or nil pointer dereferencems  0 errors  ETA 19:12
[signal 0xb code=0x1 addr=0x0 pc=0x113dfb]

goroutine 81 [running]:
panic(0x6240a0, 0xc8200140f0)
	/usr/local/Cellar/go/1.6.2/libexec/src/runtime/panic.go:481 +0x3e6
restic/archiver.(*Archiver).fileWorker(0xc820062960, 0xc886dc41f0, 0xc84181f550, 0xc84e0f3c80, 0xc82005aae0)
	/var/folders/rb/cptdcsl926bcbfyvz5thp6l80000gn/T/restic-build-122631310/src/restic/archiver/archiver.go:314 +0x16cb
created by restic/archiver.(*Archiver).Snapshot
	/var/folders/rb/cptdcsl926bcbfyvz5thp6l80000gn/T/restic-build-122631310/src/restic/archiver/archiver.go:711 +0x865
```

--- a/src/restic/archiver/archiver.go
+++ b/src/restic/archiver/archiver.go
@@ -210,14 +210,14 @@ func (arch *Archiver) SaveFile(p *restic.Progress, node *restic.Node) (*restic.N
 	file, err := fs.Open(node.Path)
 	defer file.Close()
 	if err != nil {
-		return nil, errors.Wrap(err, "Open")
+		return node, errors.Wrap(err, "Open")
 	}
 
 	debug.RunHook("archiver.SaveFile", node.Path)
 
 	node, err = arch.reloadFileIfChanged(node, file)
 	if err != nil {
-		return nil, err
+		return node, err
 	}
 
 	chnker := chunker.New(file, arch.repo.Config().ChunkerPolynomial)
@@ -230,7 +230,7 @@ func (arch *Archiver) SaveFile(p *restic.Progress, node *restic.Node) (*restic.N
 		}
 
 		if err != nil {
-			return nil, errors.Wrap(err, "chunker.Next")
+			return node, errors.Wrap(err, "chunker.Next")
 		}
 
 		resCh := make(chan saveResult, 1)
@@ -240,7 +240,7 @@ func (arch *Archiver) SaveFile(p *restic.Progress, node *restic.Node) (*restic.N
 
 	results, err := waitForResults(resultChannels)
 	if err != nil {
-		return nil, err
+		return node, err
 	}
 	err = updateNodeContent(node, results)
 

